python/tidy/servo_tidy_tests/test_tidy.py

```diff
@@ -30,6 +32,21 @@ def test_tidy_config(self):
         self.assertEqual('invalid config table [wrong]', errors.next()[2])
         self.assertNoMoreErrors(errors)
 
+    def test_directory_checks(self):
+        mock_config = {
+            "check_ext": {
+                "./python/tidy/servo_tidy_tests/dir_check/webidl_plus/": ['webidl', 'test'],
+                "./python/tidy/servo_tidy_tests/dir_check/only_webidl/": ['webidl'],
+                "./python/tidy/servo_tidy_tests/dir_check/empty/": ['webidl']
+            }
+        }
+        with mock.patch.dict('servo_tidy.tidy.config', mock_config):
+            errors = tidy.check_directory_files(False)
+            error_dir = "./python/tidy/servo_tidy_tests/dir_check/webidl_plus"
+            self.assertEqual("Unexpected extension found for test.rs. We only expect webidl, test in {0}".format(error_dir), errors.next()[2])
+            self.assertEqual("Unexpected extension found for test2.rs. We only expect webidl, test in {0}".format(error_dir), errors.next()[2])
+            self.assertNoMoreErrors(errors)
```

Comment:
Yeah, sorry, I'd totally forgotten that we're actually checking directories :smile: